packages/block-editor/src/components/block-list/use-block-props/use-block-refs.js

```diff
@@ -37,30 +36,35 @@ export function useBlockRefProvider( clientId ) {
 }
 
 /**
- * Gets a ref pointing to the current block element. Continues to return the same
- * stable ref object even if the `clientId` argument changes. This hook is not
- * reactive, i.e., it won't trigger a rerender of the calling component if the
- * ref value changes. For reactive use cases there is the `useBlockElement` hook.
+ * Tracks the DOM element for the block identified by `clientId` and assigns it to the `ref`
+ * whenever it changes.
  *
- * @param {string} clientId The client ID to get a ref for.
- *
- * @return {RefObject} A ref containing the element.
+ * @param {string} clientId The client ID to track.
+ * @param {Ref}    ref      The ref object/callback to assign to.
  */
-function useBlockRef( clientId ) {
+export function useBlockElementRef( clientId, ref ) {
```

Comment:
You mean, the refs would be stored directly in the `refsMap` instead of the element? And the `refsMap` job would be to "forward" the ref callback calls to the registered consumers?

I think in that case it would be hard to register a new ref at a time when the block is already mounted. When calling:
```js
useBlockElementRef( clientId, ref );
```
The `ref` wouldn't be set until the block element is changed or recreated.